From 2ea7588c18ef673137062804b1ecb50e32748ae9 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Tue, 21 Nov 2023 23:39:48 +0900
Subject: FIX: fix build error with libxml2 2.12.0

libxml2 2.12.0 changed which header file to define functions
and header inclusion. Due to this refactoring, some yaz source
files need additional inclusion of libxml2 header file,
especially for libxml/parser.h .

Closes #102 .

diff --git a/client/client.c b/client/client.c
index 34e6a77..369564a 100644
--- a/client/client.c
+++ b/client/client.c
@@ -73,6 +73,10 @@
 #include <readline/history.h>
 #endif
 
+#if YAZ_HAVE_XML2
+#include <libxml/parser.h>
+#endif
+
 
 #include "admin.h"
 #include "tabcomplete.h"
diff --git a/src/record_render.c b/src/record_render.c
index 82fa698..654aca4 100644
--- a/src/record_render.c
+++ b/src/record_render.c
@@ -23,6 +23,7 @@
 #include <yaz/base64.h>
 
 #if YAZ_HAVE_XML2
+#include <libxml/parser.h>
 #include <libxml/xpath.h>
 #include <libxml/xpathInternals.h>
 #endif
diff --git a/test/test_ccl.c b/test/test_ccl.c
index bb3e932..b970870 100644
--- a/test/test_ccl.c
+++ b/test/test_ccl.c
@@ -11,6 +11,9 @@
 #include <yaz/log.h>
 #include <yaz/test.h>
 
+#if YAZ_HAVE_XML2
+#include <libxml/parser.h>
+#endif
 
 static int tst_ccl_query(CCL_bibset bibset,
                          const char *query,
diff --git a/test/test_icu.c b/test/test_icu.c
index 18e2866..309621a 100644
--- a/test/test_icu.c
+++ b/test/test_icu.c
@@ -28,6 +28,7 @@
 #endif
 
 #if YAZ_HAVE_XML2
+#include <libxml/parser.h>
 #include <libxml/xmlmemory.h>
 #endif
 
diff --git a/test/test_xml_include.c b/test/test_xml_include.c
index aec8308..ff1f1a1 100644
--- a/test/test_xml_include.c
+++ b/test/test_xml_include.c
@@ -12,6 +12,9 @@
 
 #include <yaz/xml_include.h>
 #include <yaz/test.h>
+#if YAZ_HAVE_XML2
+#include <libxml/parser.h>
+#endif
 
 static void tst_xml_include1(void)
 {
diff --git a/util/cclsh.c b/util/cclsh.c
index 419fcb7..c40061c 100644
--- a/util/cclsh.c
+++ b/util/cclsh.c
@@ -20,6 +20,9 @@
 #include <readline/history.h>
 #endif
 
+#if YAZ_HAVE_XML2
+#include <libxml/parser.h>
+#endif
 
 static int debug = 0;
 static char *prog;
diff --git a/util/yaz-icu.c b/util/yaz-icu.c
index 3087e13..767cb20 100644
--- a/util/yaz-icu.c
+++ b/util/yaz-icu.c
@@ -17,6 +17,8 @@
 
 #if YAZ_HAVE_ICU
 
+#include <libxml/parser.h>
+
 #include <unicode/ucnv.h>
 #include <unicode/ustring.h>
 #include <unicode/ucol.h>
diff --git a/util/yaz-record-conv.c b/util/yaz-record-conv.c
index 75be066..b386e2a 100644
--- a/util/yaz-record-conv.c
+++ b/util/yaz-record-conv.c
@@ -13,6 +13,10 @@
 #include <yaz/record_conv.h>
 #include <yaz/backtrace.h>
 
+#if YAZ_HAVE_XML2
+#include <libxml/parser.h>
+#endif
+
 const char *prog = "yaz-record-conv";
 
 static void usage(void)
-- 
2.42.1


From 5e57b3aa8a468e3cc8ff56795a8448021d1f3f8a Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Tue, 21 Nov 2023 23:47:18 +0900
Subject: FIX: fix build error with glibc 2.39

glibc 2.39 does some refactoring for header file inclusion
and some additional header inclusion is needed for yaz
source.

Closes #104 .

diff --git a/src/record_conv.c b/src/record_conv.c
index 6ec066f..6daaa9a 100644
--- a/src/record_conv.c
+++ b/src/record_conv.c
@@ -11,6 +11,7 @@
 #include <config.h>
 #endif
 
+#include <stdlib.h>
 #include <string.h>
 #include <yaz/log.h>
 #include <yaz/yaz-iconv.h>
diff --git a/src/xmlquery.c b/src/xmlquery.c
index 543be2a..dc2bfd3 100644
--- a/src/xmlquery.c
+++ b/src/xmlquery.c
@@ -10,6 +10,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <assert.h>
 
diff --git a/test/test_record_conv.c b/test/test_record_conv.c
index bb417a5..7fdca22 100644
--- a/test/test_record_conv.c
+++ b/test/test_record_conv.c
@@ -9,6 +9,7 @@
 #include <yaz/record_conv.h>
 #include <yaz/test.h>
 #include <yaz/wrbuf.h>
+#include <stdlib.h>
 #include <string.h>
 #include <yaz/log.h>
 #include <yaz/proto.h>
diff --git a/test/test_retrieval.c b/test/test_retrieval.c
index 25711af..6065c5c 100644
--- a/test/test_retrieval.c
+++ b/test/test_retrieval.c
@@ -9,6 +9,7 @@
 #include <yaz/retrieval.h>
 #include <yaz/test.h>
 #include <yaz/wrbuf.h>
+#include <stdlib.h>
 #include <string.h>
 #include <yaz/log.h>
 #include <yaz/oid_db.h>
-- 
2.42.1

